src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/MtAmazonDynamoDbBySharedTable.java

```diff
@@ -414,11 +416,20 @@ public GetItemResult getItem(GetItemRequest getItemRequest) {
         return getItemResult;
     }
 
-    TableMapping getTableMapping(String virtualTableName) {
+    Optional<TableMapping> getTableMapping(String virtualTableName) {
         try {
-            return tableMappingCache.get(virtualTableName, () ->
+            return Optional.of(tableMappingCache.get(virtualTableName, () ->
                 tableMappingFactory.getTableMapping(
-                    new DynamoTableDescriptionImpl(mtTableDescriptionRepo.getTableDescription(virtualTableName))));
+                    new DynamoTableDescriptionImpl(mtTableDescriptionRepo.getTableDescription(virtualTableName)))));
+        } catch (UncheckedExecutionException e) {
+            // This isn't great, but we're assuming a missing virtual table entry here is a deleted virtual
+            // table, and thus, shouldn't be mapping records to MtRecords. Instead return null, but really,
+            // we should make sure we let deleted records flow through a stream and expire out, before removing
+            // the virtual table entry.
+            if (e.getCause() instanceof ResourceNotFoundException) {
+                return Optional.empty();
```

Comment:
Should we cache empty results?

Reply:
ah good catch, will cause a lot of hitting of raw stream if we don't cache the delete markers

Reply:
hitting of the raw description table **